engine/runtime-integration-tests/src/test/scala/org/enso/compiler/test/core/ir/MetadataStorageTest.scala

```diff
@@ -200,8 +201,8 @@ class MetadataStorageTest extends CompilerTest {
         )
       )
 
-      meta.duplicate shouldEqual meta
-      meta.duplicate shouldEqual expected
+      meta.duplicate shouldNot equal(meta)
```

Comment:
How comes `duplicate` isn't equal to original? I do understand why it is not the same reference (e.g. not Java `==`), but why it is not `Object.equals`?

Reply:
Ááá, it is because of different `MetadataStorage()` and `Diagnostics`, right? They are `Object.equals` only when references are `==` and `duplicate` makes sure the instances are different.

Not sure if it makes sense to compare `IR.Module.equals(anotherIrModule)` then. Either it is reference `==` or it is going to be (almost certainly) different....

Reply:
Yes, I don't think it makes sense.